fth/system.fth

```diff
@@ -596,6 +596,29 @@ ustack 0stackp
         char [compile] literal
 ; immediate
 
+: TOUPPER ( char -- char' )
+	dup [char] a >=
+	IF
+		dup [char] z <= IF [ char A char a - ] literal + THEN
+	THEN
+;
+
+: UPCASE ( c-addr u -- )
```

Comment:
Using addresses as loop indices feels a bit odd. How about this:


: UPCASE ( c-addr count -- , converts a string to upper case in place )
    0 ?DO
        dup c@ toupper over c! char+
    LOOP drop
;